packages/commonjs/src/dynamic-packages-manager.js

```diff
@@ -34,17 +40,29 @@ export function getDynamicPackagesEntryIntro(
 ) {
   let dynamicImports = Array.from(
     dynamicRequireModuleSet,
-    (dynamicId) => `require(${JSON.stringify(DYNAMIC_REGISTER_PREFIX + dynamicId)});`
+    (dynamicId) => `require(${JSON.stringify(wrapModuleRegisterProxy(dynamicId))});`
   ).join('\n');
 
   if (dynamicRequireModuleDirPaths.length) {
-    dynamicImports += `require(${JSON.stringify(DYNAMIC_REGISTER_PREFIX + DYNAMIC_PACKAGES_ID)});`;
+    dynamicImports += `require(${JSON.stringify(wrapModuleRegisterProxy(DYNAMIC_PACKAGES_ID))});`;
   }
 
   return dynamicImports;
 }
 
-export function isModuleRegistrationProxy(id, dynamicRequireModuleSet) {
+export function wrapModuleRegisterProxy(id) {
+  return DYNAMIC_REGISTER_PREFIX + id;
+}
+
+export function unwrapModuleRegisterProxy(id) {
+  return id.slice(DYNAMIC_REGISTER_PREFIX.length);
+}
+
+export function isModuleRegisterProxy(id) {
+  return id.startsWith(DYNAMIC_REGISTER_PREFIX);
+}
+
+export function isDynamicModuleImport(id, dynamicRequireModuleSet) {
```

Comment:
If we were to change the prefix to be a suffix instead, we would not need these but could reuse `wrapId`, `unwrapId` and `isWrappedId` from `helpers.js`. What do you think?

Reply:
I tried that- but it caused several issues. I'll try again if I have time today

Reply:
If it does not work out, no problem. Was just wondering because it looked kind of symmetric. Also, there is one thing I am worried about. Let's assume we have an id `foo`:

```js
const id = "foo";
const wrapped = wrapId(id, REQUIRE_SUFFIX); // \0foo?commonjs-require
const doubleWrapped = wrapModuleRegisterProxy(wrapped); // \0commonjs-dynamic-register:\0foo?commonjs-require

console.log(isWrappedId(doubleWrapped, REQUIRE_SUFFIX)); // true, because we are only looking at the suffix
const unwrapped = unwrapId(doubleWrapped, REQUIRE_SUFFIX)); // commonjs-dynamic-register:\0foo

console.log(isModuleRegisterProxy(unwrapped)); // false!!
```

So if we first wrap with a suffix and then a prefix, the `isWrappedId` check for the suffix is still fulfilled, but unwrapping the suffix first will end up with a broken id.

My feeling is we should rather switch to suffixes in all cases where there could be conflict and make sure we first unwrap properly whenever this causes issues to avoid these potentially unnoticed issues.

Reply:
Yeah the thing is I actually should make the transition now anyway, as it messes up with `node-resolve` (importer has a prefix it can't understand) so try-catch tests are now failing with this. I can either strip the importer from prefixes, or move to suffixes which seem more reasonable.

Reply:
Okay so `extname(id)` in `transform` callback was what was stopping it from working previously. Understandably :-)

Reply:
Done